CMakeLists.txt

```diff
@@ -540,6 +540,7 @@ if(SEAL_BUILD_SEAL_C)
     seal_set_version(sealc)
     if(SEAL_BUILD_STATIC_SEAL_C)
         seal_set_version_filename(sealc)
+        add_compile_definitions(SEAL_BUILD_STATIC_SEAL_C)
```

Comment:
@rickwebiii I'm not familiar with this command. What does it do?

Reply:
It passes a `#define` to the C++ compiler. In gcc/clang, this is basically the `-D` flag.

Reply:
Does it work for a downstream app? It cannot see `#define SEAL_BUILD_STATIC_SEAL_C`. So when a downstream app includes `seal/c/batchencoder.h` that includes `seal/c/defines.h`, `__declspec(dllimport)` is always used, right?

Reply:
Oh, good point. Users would have to add this line to their CMAKE file or some equivalent, which I agree isn't great. I'll do some investigation and try to find how to make libraries both statically and dynamically linkable in Windows.

Reply:
I cannot approve this PR right now. It's really useful though. Shall we keep this PR open? We will release a new version of SEAL soon. We can always rebase this PR to a new version -- there shouldn't be hard conflicts.

Reply:
I need to test this, but it looks like you can always omit `declspec(dllimport)` and the headers would work for both static and dynamic linking.

The [cost](https://docs.microsoft.com/en-us/cpp/build/importing-function-calls-using-declspec-dllimport?view=msvc-170) of doing so appears to be a single `jmp` instruction (a few clock cycles?), which is probably fine for the vast majority of calls. `dllimport` allows you to avoid a thunk in many cases. You only incur this cost with dynamic linking; calling a statically linked function doesn't require a thunk.